src/Allegro.Extensions.Cqrs/Allegro.Extensions.Cqrs.Abstractions/Queries/Query.cs

```diff
@@ -0,0 +1,19 @@
+using System;
+
+namespace Allegro.Extensions.Cqrs.Abstractions.Queries;
+
+/// <summary>
+/// Cqrs base representation of query.
+/// </summary>
+public abstract record Query
+{
+    /// <summary>
+    /// Query identifier
+    /// </summary>
+    public string Id { get; } = Guid.NewGuid().ToString();
+}
+
+/// <summary>
+/// Cqrs base representation of query.
+/// </summary>
```

Comment:
nit: missing description of the `T` type param

Reply:
I guess it should be
```suggestion
<typeparam name="TResult">Type of data returned by query</typeparam>
```

Reply:
fixed